src/test/java/com/fasterxml/jackson/core/read/NonStandardNumberParsingTest.java

```diff
@@ -115,6 +115,12 @@ private void _testLeadingDotInDecimalAllowed(JsonFactory f, int mode) throws Exc
             assertEquals("0.125", p.getDecimalValue().toString());
             assertEquals(".125", p.getText());
         }
+        try (JsonParser p = createParser(f, mode, " -.125 ")) {
+            assertEquals(JsonToken.VALUE_NUMBER_FLOAT, p.nextToken());
+            assertEquals(-0.125, p.getValueAsDouble());
```

Comment:
I think there is also `getDoubleValue()`, and variants with `Float`.

Reply:
this is just a copy of the previous few lines but testing a different input - the issue is not with me using the wrong method - the jackson code has a bug (in my opinion) that leads to it ignoring the sign on '-.125'

Reply:
Yes, I realized that. What I mean is just that test itself should probably also cover other accessors -- but did not mean to imply that was causing problems or relevant to the actual problem. Methods used are fine, just a subset of all available. So for now we can ignore that.